File: Segundo/PDOO/CivitasJava/src/main/java/civitas/CasillaCalle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author jmramirez
 */
public class CasillaCalle extends Casilla {
    private TituloPropiedad titulo;
    
    CasillaCalle(TituloPropiedad tit) {
        super(tit.getNombre());
        titulo = tit;
    }
    
    void recibeJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            Jugador jugador = todos.get(actual);
            if (!titulo.tienePropietario()) {
                jugador.puedeComprarCasilla();                
            } else {
                titulo.tramitarAlquiler(jugador);
            }
        }
    }
    
    public String toString() {
        String ret = "la casilla ";
        ret += nombre;
        ret += ", de tipo Calle ";
        if (!titulo.tienePropietario()) {
            ret += "con un precio de ";
            ret += titulo.getPrecioCompra();
        }
        ret += ".";
        return ret;
    }
    
    TituloPropiedad getTituloPropiedad() {
        return titulo;
    }
}

File: Segundo/PDOO/CivitasRuby/Civitas/CasillaCalle.rb
# encoding:utf-8

require_relative "Casilla"
require_relative "TituloPropiedad"

module Civitas

    class CasillaCalle < Casilla
        
        attr_reader :titulo
        attr_reader :cantidad
        
        def initialize(tit)
            super(tit.nombre)
            @titulo = tit
            @cantidad = tit.precioCompra
        end

        def informe(actual, todos)
            if !@titulo.tienePropietario()
                Diario.instance.ocurre_evento("El jugador " + todos.at(actual).nombre + " ha caído en una casilla de tipo Calle, con un precio de " + @cantidad.to_s + ".")
            else
                Diario.instance.ocurre_evento("El jugador " + todos.at(actual).nombre + " ha caído en una casilla de tipo Calle.")
            end
        end

        def recibe_jugador(actual, todos)
            if jugadorCorrecto(actual, todos)
                informe(actual, todos)
                jugador = todos.at(actual)
                if !@titulo.tienePropietario()
                    jugador.puedeComprarCasilla()
                else
                    @titulo.tramitarAlquiler(jugador)
                end
            end
        end

        def toString
            ret = "La casilla actual es " + @nombre + " y es de tipo Calle."
            if !titulo.tienePropietario()
                ret += " Además, tiene un precio de " + @cantidad.to_s + "."
            end
            ret += "\n"
            return ret
        end

    end

end